src/device/mod.rs

```diff
@@ -750,9 +750,9 @@ impl Device {
                                 eprintln!("Fatal error on tun interface: errno {:?}", errno);
                                 return Action::Exit;
                             }
-                            continue;
+                            break;
                         }
-                        Err(_) => continue,
+                        Err(_) => break,
```

Comment:
Probably should eprint this error too?

Reply:
I think it makes sense. I would even suggest to stop the handler altogether if this occurs.

From what I understand (looking at both `tun_darwin.rs` and `tun_linux.rs`), `read` should only return `Error::IfaceRead` errors. Any other kind of error is unexpected and that might be a good reason to stop the application.

I made a tentative change, where any error different from `IfaceRead` is considered fatal also.

On the other hand, maybe we should change the `read()` return error type to `Error::IfaceRead` only and remove this match arm altogether?

Reply:
I really want to change all errors to be std::io::Error in the future